sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/trace/sleuth/AzureHttpClientBuilderFactoryBeanPostProcessor.java

```diff
@@ -38,12 +38,13 @@ public void setBeanFactory(BeanFactory beanFactory) {
     @SuppressWarnings({ "rawtypes", "unchecked" })
     public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
         if (bean instanceof AbstractAzureHttpClientBuilderFactory) {
-            HttpPipelinePolicy policy = beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME, HttpPipelinePolicy.class);
-            if (policy != null) {
+            try {
+                HttpPipelinePolicy policy = beanFactory.getBean(DEFAULT_SLEUTH_HTTP_POLICY_BEAN_NAME,
+                    HttpPipelinePolicy.class);
                 AbstractAzureHttpClientBuilderFactory builderFactory = (AbstractAzureHttpClientBuilderFactory) bean;
                 builderFactory.addHttpPipelinePolicy(policy);
                 LOGGER.debug("Added the Sleuth http pipeline policy to {} builder.", bean.getClass());
-            } else {
+            } catch (BeansException exception) {
```

Comment:
Why don't we use beanFactory.containsBean here?

Reply:
Use bean name and bean class name safer, is it necessary?

Reply:
I don't get your question here.